packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -50,11 +55,26 @@ export interface VScopedSlotFragment extends BaseVNode {
 }
 
 export interface VStaticPart {
+    readonly type: VStaticPartType;
     readonly partId: number;
+    readonly data?: VStaticPartData;
+    readonly text?: string;
+    elm: Element | Text | undefined;
+}
+
+export interface VStaticPartElement extends VStaticPart {
+    readonly type: VStaticPartType.Element;
     readonly data: VStaticPartData;
     elm: Element | undefined;
 }
+
+export interface VStaticPartText extends VStaticPart {
+    readonly type: VStaticPartType.Text;
```

Comment:
This `type` is not strictly necessary, correct? Can we duck-type this based on the presence of the `text`? (I only mention this because it might be a slight perf boost in terms of sending less data, doing fewer accesses on the object.)

Reply:
On second thought, the `type` is fine. It makes the code more readable.